rpcserver.go

```diff
@@ -3459,34 +3458,38 @@ func UnmarshalOutpoint(outpoint string) (*wire.OutPoint, error) {
 // unmarshalLeafKey un-marshals a leaf key from the RPC form.
 func unmarshalLeafKey(key *unirpc.AssetKey) (universe.LeafKey, error) {
 	var (
-		leafKey universe.LeafKey
-		err     error
+		leafKey        universe.LeafKey
+		scriptKey      *btcec.PublicKey
+		scriptKeyBytes []byte
+		outpoint       *wire.OutPoint
+		hash           *chainhash.Hash
+		err            error
 	)
 
 	switch {
 	case key.GetScriptKeyBytes() != nil:
-		pubKey, err := parseUserKey(key.GetScriptKeyBytes())
+		scriptKey, err = parseUserKey(key.GetScriptKeyBytes())
 		if err != nil {
 			return leafKey, err
 		}
 
 		leafKey.ScriptKey = &asset.ScriptKey{
-			PubKey: pubKey,
+			PubKey: scriptKey,
 		}
 
 	case key.GetScriptKeyStr() != "":
-		scriptKeyBytes, sErr := hex.DecodeString(key.GetScriptKeyStr())
-		if sErr != nil {
+		scriptKeyBytes, err = hex.DecodeString(key.GetScriptKeyStr())
```

Comment:
I guess this is the only problem that we should really fix, since before this change we assigned `sErr` but returned `err`.